plutus-use-cases/test/Spec/Auction.hs

```diff
@@ -241,3 +221,27 @@ finishAuction = do
 prop_FinishAuction :: Property
 prop_FinishAuction = forAllDL finishAuction prop_Auction
 
+tests :: TestTree
+tests =
+    testGroup "auction"
+        [ checkPredicateOptions options "run an auction"
+            (assertDone seller (Trace.walletInstanceTag w1) (const True) "seller should be done"
+            .&&. assertDone buyer (Trace.walletInstanceTag w2) (const True) "buyer should be done"
+            .&&. assertAccumState buyer (Trace.walletInstanceTag w2) ((==) (Just $ Last trace1FinalState)) "final state should be OK"
+            .&&. walletFundsChange w1 (Ada.toValue trace1WinningBid <> inv theToken)
+            .&&. walletFundsChange w2 (inv (Ada.toValue trace1WinningBid) <> theToken))
+            auctionTrace1
+        , checkPredicateOptions options "run an auction with multiple bids"
+            (assertDone seller (Trace.walletInstanceTag w1) (const True) "seller should be done"
+            .&&. assertDone buyer (Trace.walletInstanceTag w2) (const True) "buyer should be done"
+            .&&. assertDone buyer (Trace.walletInstanceTag w3) (const True) "3rd party should be done"
+            .&&. assertAccumState buyer (Trace.walletInstanceTag w2) ((==) (Just $ Last trace2FinalState)) "final state should be OK"
+            .&&. walletFundsChange w1 (Ada.toValue trace2WinningBid <> inv theToken)
+            .&&. walletFundsChange w2 (inv (Ada.toValue trace2WinningBid) <> theToken)
+            .&&. walletFundsChange w3 mempty)
+            auctionTrace2
+        -- FIXME: Fails because you cannot bid in certain slots when the off-chain code is busy.
```

Comment:
Seems like we should fix it!

Reply:
Or is this a bug in our stuff?

Reply:
Yeah I expected it to be fixed in the next two weeks (has to do with waiting vs non-waiting actions)